app/config/errors.php

```diff
@@ -10,61 +10,64 @@
         'name' => Exception::GENERAL_UNAUTHORIZED_SCOPE,
-        'description' => 'The current user or API key does not have the required scopes to access the requested resource.',
+        'description' => 'The current user or API key does not have the required scopes to access the requested resource. Update the permissions of the requested resource or add appropriate scopes to your API key.',
         'code' => 401,
     ],
     Exception::GENERAL_RATE_LIMIT_EXCEEDED => [
         'name' => Exception::GENERAL_RATE_LIMIT_EXCEEDED,
-        'description' => 'Rate limit for the current endpoint has been exceeded. Please try again after some time.',
+        'description' => 'Rate limit for the current endpoint has been exceeded. Please try again after some time. Learn more about rate-limits here https://appwrite.io/docs/rate-limits.',
         'code' => 429,
     ],
     Exception::GENERAL_SMTP_DISABLED => [
         'name' => Exception::GENERAL_SMTP_DISABLED,
-        'description' => 'SMTP is disabled on your Appwrite instance. You can <a href="/docs/email-delivery">learn more about setting up SMTP</a> in our docs.',
+        'description' => 'Email delivery is not possible because SMTP is disabled on your Appwrite instance. You can learn more about setting up SMTP here: https://appwrite.io/docs/email-delivery.',
         'code' => 503,
     ],
     Exception::GENERAL_PHONE_DISABLED => [
         'name' => Exception::GENERAL_PHONE_DISABLED,
-        'description' => 'Phone provider is not configured. Please check the _APP_SMS_PROVIDER environment variable of your Appwrite server.',
+        'description' => 'SMS delivery is not possible because a phone provider is not configured. You can learn more about setting up SMS delivery here: https://appwrite.io/docs/sms-delivery.',
         'code' => 503,
     ],
     Exception::GENERAL_ARGUMENT_INVALID => [
         'name' => Exception::GENERAL_ARGUMENT_INVALID,
-        'description' => 'The request contains one or more invalid arguments. Please refer to the endpoint documentation.',
+        // TODO: This is used by the DB, can we make this more specific instead of a generic error or add some kind of templating into the message?
+        'description' => 'The argument {argument} is invalid. Please refer to the endpoint documentation {docs}.',
         'code' => 400,
     ],
     Exception::GENERAL_QUERY_LIMIT_EXCEEDED => [
         'name' => Exception::GENERAL_QUERY_LIMIT_EXCEEDED,
+        // TODO: Would it be possible to indicate which attribute? It is really hard to debug when it just says "current attribute"
         'description' => 'Query limit exceeded for the current attribute. Usage of more than 100 query values on a single attribute is prohibited.',
         'code' => 400,
     ],
     Exception::GENERAL_QUERY_INVALID => [
         'name' => Exception::GENERAL_QUERY_INVALID,
+        // TODO: Can we provide more detailed feedback about which part of the query is invalid? 
```

Comment:
How do we parse queries? Is it possible to indicate exactly which part of a query is invalid? This message is vague.

Reply:
This is also not currently used.

<img width="1006" alt="image" src="https://github.com/appwrite/appwrite/assets/1477010/b9ad6ec2-33b6-4155-8a4f-46b4660807e9">

Search: https://github.com/search?q=repo%3Aappwrite%2Fappwrite+GENERAL_QUERY_INVALID&type=code

Reply:
Okay cooolllll. Should this exist then?

Reply:
@gewenyu99 similar to my comment above. It was used when it was initially implemented. But it has then been moved to the queries validator. Lets check with @abnegate